src/ResultWriter/WaveFieldWriter.cpp

```diff
@@ -285,7 +296,16 @@ void seissol::writer::WaveFieldWriter::init(unsigned int numVars,
 
 		// Create data buffers
 		param.bufferIds[LOWVARIABLE0] = addBuffer(0L, pLowMeshRefiner->getNumCells() * sizeof(double));
-		for (unsigned int i = 1; i < WaveFieldWriterExecutor::NUM_LOWVARIABLES; i++)
+		int numLowVars = 0;
+		if (pstrain && !integrals) {
+			numLowVars = WaveFieldWriterExecutor::NUM_LOWVARIABLES;
+		} else if (integrals && !pstrain) {
+			numLowVars = m_numIntegratedVariables;
+		} else {
+			numLowVars = WaveFieldWriterExecutor::NUM_LOWVARIABLES + m_numIntegratedVariables;
+		}
+
+		for (unsigned int i = 1; i < numLowVars; i++)
```

Comment:
You can write this with:
`numLowVars = 0`
`if (pstrain) numLowVars += WaveFieldWriterExecutor::NUM_LOWVARIABLES;`
`if (integrals) numLowVars += m_numIntegratedVariables;`